src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -1007,15 +1023,39 @@ void ResumptionDataProcessor::DeletePluginsSubscriptions(
   }
 
   const ApplicationResumptionStatus& status = it->second;
-  smart_objects::SmartObject extension_subscriptions;
+  smart_objects::SmartObject extension_vd_subscriptions;
   for (auto ivi : status.successful_vehicle_data_subscriptions_) {
     LOG4CXX_DEBUG(logger_, "ivi " << ivi << " should be deleted");
-    extension_subscriptions[ivi] = true;
+    extension_vd_subscriptions[ivi] = true;
+  }
+
+  smart_objects::SmartObject extension_modules_subscriptions(
+      smart_objects::SmartType_Map);
+
+  if (!status.succesfull_module_subscriptions_.empty()) {
+    extension_modules_subscriptions[message_params::kModuleData] =
+        new smart_objects::SmartObject(smart_objects::SmartType_Array);
+
+    auto& module_data_so =
+        extension_modules_subscriptions[message_params::kModuleData];
+
+    uint32_t index = 0;
+    for (const auto& module : status.succesfull_module_subscriptions_) {
+      module_data_so[index] =
+          new smart_objects::SmartObject(smart_objects::SmartType_Map);
+      module_data_so[index][message_params::kModuleType] = module.first;
+      module_data_so[index][message_params::kModuleId] = module.second;
+      index++;
+    }
+
+    extension_vd_subscriptions[message_params::kModuleData] =
+        extension_modules_subscriptions[message_params::kModuleData];
   }
   resumption_status_lock_.Release();
 
-  for (auto& extension : application->Extensions()) {
-    extension->RevertResumption(extension_subscriptions);
+  auto extenstions = application->Extensions();
+  for (auto& extension : extenstions) {
+    extension->RevertResumption(extension_vd_subscriptions);
```

Comment:
@jacobkeeler  In the current implementation, `ResumptioDataProcessor` doesn't know about any application extensions exactly, just by application extension interface.
We gether all resumption data for each application extension in one SmartObject, that are in seperate sections. And each application extension knows only about its own data section and retirieves only its own resumption data.
Also, we can't separate each application extension by ID, because the ID of each application extension is declared in each application extension file, and in this case, we should include all application extension files into the `ResumptionDataProcessor` and we would like to avoid that.

Reply:
@IGapchuk Alright, I would at least make sure that the resumption fields are separated out accordingly. It looks like the vehicle data subscriptions are in the root level of the object while the interior VD subscriptions are contained under the `moduleData` key. I think both groups of subscriptions should be in separate sections of the object. Something like

```json
{
  "VehicleInfo": {
    "gps": true,
    "speed": true,
    "prndl": true
  },
  "RemoteControl": {
    "moduleData": [
      {
        "moduleType": "CLIMATE",
        "moduleID": "climate-module-no-1"
      },
      {
        "moduleType": "SEAT",
        "moduleID": "seat-module-no-1"
      }
    ]
  }
}
```

Reply:
@jacobkeeler Please take a look at the 1e444d0c8ca14fa371febba45dd4f3561dc131f6. Now resumption data to revert for each extension save in the separate section.